### PR TITLE
Bump py compat target to 3.8 in setup.py and CI

### PR DESCRIPTION
**Pull request by inducer**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,4 +34,4 @@ test-command = "pytest {project}/test/test_array.py" # same limitation as conda-
 skip = ["*-win32", "pp*"]
 test-command = ""
 before-all = "bash {package}/scripts/build-ocl-windows.sh"
-before-build = "python configure.py --cxxflags=-ID:/a/pyopencl/pyopencl/OpenCL-Headers/install/include --ldflags=\"/LIBPATH:C:/Program Files/OpenCL-ICD-Loader/lib\""
+before-build = "python configure.py --cxxflags=-ID:/a/pyopencl/pyopencl/OpenCL-Headers/install/include --cl-lib-dir=\"C:/Program Files/OpenCL-ICD-Loader/lib\""
```
Comment (isuruf) [inline pyproject.toml]: Use `--cl-inc-dir` too?
```suggestion
before-build = "python configure.py --cl-inc-dir=D:/a/pyopencl/pyopencl/OpenCL-Headers/install/include --cl-lib-dir=\"C:/Program Files/OpenCL-ICD-Loader/lib\""
```

Comment (inducer) [reply pyproject.toml]: Thanks! I thought about it, but, in retrospect, I would have guessed wrong in how to do it. :) Not that it's not obvious in hindsight. I just don't know Windows.

**Comment by inducer** [comment]
The Windows build is still unhappy at the link stage (the recent include dir change notwithstanding):
```
        wrap_cl_part_1.obj : error LNK2001: unresolved external symbol _clCreateBuffer@24
        wrap_cl_part_1.obj : error LNK2001: unresolved external symbol _clEnqueueCopyBufferRect@52
        wrap_cl_part_1.obj : error LNK2001: unresolved external symbol _clReleaseCommandQueue@4
        wrap_cl_part_1.obj : error LNK2001: unresolved external symbol _clReleaseProgram@4
```

despite receiving ` "/LIBPATH:C:/Program Files/OpenCL-ICD-Loader/lib"` and `OpenCL.lib` on  the command line.

https://github.com/inducer/pyopencl/actions/runs/3509660036/jobs/5878911594

I don't know whether the `@N` suffixes are to blame, or where they might come from.

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -26,7 +26,9 @@ jobs:
           submodules: 'true'
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@v2.11.1
+        uses: pypa/cibuildwheel@v2.11.2
+        env:
+            CIBW_SKIP: cp36-* cp37-* pp37-*
```
Comment (isuruf) [inline .github/workflows/wheels.yml]: Setting this ignores the skipping in https://github.com/inducer/pyopencl/blob/main/pyproject.toml#L34 which results in cibw building `cp38-win32` builds (windows i686 builds)

Comment (inducer) [reply .github/workflows/wheels.yml]: Thanks! In retrospect, I could have figured this out. Thanks for the help!

Comment (isuruf) [reply .github/workflows/wheels.yml]: :smile: It took me an embarrassing amount of time and a tmate session to figure out that we were building for i686.

Comment (inducer) [reply .github/workflows/wheels.yml]: You can tmate into Windows CI runners?!

Comment (isuruf) [reply .github/workflows/wheels.yml]: Yes. It drops you into a msys2 shell.

Comment (inducer) [reply .github/workflows/wheels.yml]: That's super useful to know, and it makes me less averse to (at least trying to) support Windows.

**Pull request merged**